meeg_preprocessing/utils.py

```diff
@@ -151,15 +153,28 @@ def setup_provenance(script, results_dir, config=None, use_agg=True):
     - sets log file for sterr output
     - writes log file with runtime information
     """
+    import inspect
+    if script is None:
+        script = inspect.stack()[1][1]
+        if script[0] == '<':
+            script = '__main__'
```

Comment:
Never trust cats

ok. So you'd be ok for a default script=False & results_dir=False that skips the log of script but setups the rest of the report?